kongswap_adaptor/src/canister.rs

```diff
@@ -205,16 +210,30 @@ impl<A: AbstractAgent> TreasuryManager for KongSwapAdaptor<A> {
     }
 
     async fn issue_rewards(&mut self) {
-        if let Err(err) = self.check_state_lock() {
-            log_err(&format!("Cannot issue rewards: {:?}", err));
-            return;
-        }
+        // This operation _can_ and _should_ be lock-free.
```

Comment:
@aterga I thought there was potentially an edge case where you could be issuing rewards during a deposit (so that the balance was increased during the deposit, but the rest of the transaction wasn't finished, so that this function would believe there were rewards to issue by looking at the balances?)

Reply:
I think it was a question about when the balances are updated... and maybe that was the crux of the conversation - so long as balances are only updated in deposit after the deposit is successful (rather than at the beginning of the call, which would allow for interleavings) then this function can't do an incorrect calculation about what should be sent to the treasury.